src/language/from_relay/mod.rs

```diff
@@ -1808,4 +1813,306 @@ def @main(%x: Tensor[(3), float32]) -> Tensor[(3), float32] {
+                );
+
+                // TODO(@gussmith23) output type assumption
+                let relay_output: ndarray::ArrayD<f32> = read_npy(output_filepath).unwrap();
+
+                b.iter(|| {
+                    // use black box to prevent compiler optimizations
+                    let expr = test::black_box(&expr);
+                    let env = test::black_box(&env);
+
+                    let interpreter_output = match interpret(&expr, expr.as_ref().len() - 1, &env) {
+                        crate::language::interpreter::Value::Access(a) => a.tensor,
+                        _ => panic!(),
+                    };
+
+                    // TODO: this was a hack because NAN != NAN
+                    // relay_output.mapv_inplace(|x| if x.is_nan() { 0.0 } else { x });
+                    // interpreter_output.mapv_inplace(|x| if x.is_nan() { 0.0 } else { x });
+                    print!("{:?}\nvs.\n{:?}", relay_output, interpreter_output);
+                    assert!(
+                        relay_output.abs_diff_eq(&interpreter_output, 1e-5),
+                        "{:?}\nvs.\n{:?}",
+                        relay_output,
+                        interpreter_output
+                    );
+                });
+            }
+        };
+    }
+
+    // note: mobilenet-shallow was generated:
+    // conv_block_1, separable_conv_block_1, separable_conv_block_2, and the layers at the end
+    // i.e all the conv_blocks from separable_conv_block_3 onward were removed
+    //
+    // tvm.relay.testing.mobilenet.mobile_net(num_classes = 10, data_shape=(1,3,32,32))
+    // followed by SimplifyInference
+    benchmark_model!(
+        mobilenet_shallow,
+        format!(
+            "{}/models/mobilenet-shallow.relay",
+            env!("CARGO_MANIFEST_DIR")
+        )
+    );
+
+    // note: resnet-shallow was generated:
+    // tvm.relay.testing.resnet.get_workload(num_layers=2, num_classes=10, image_shape=(3,28,28))
+    // followed by SimplifyInference
+    benchmark_model!(
+        resnet_shallow,
+        format!("{}/models/resnet-shallow.relay", env!("CARGO_MANIFEST_DIR"))
+    );
+
+    // TODO: TESTS ARE IGNORED BECAUSE INTERPRETER TOO SLOW
+    // ====================================================
+    // Creates a Relay-to-Glenside test over a model
+    // (currently the models we test are convolutional and only image-based)
+    // The test does the following:
+    //  1. Reads the $relay_str_path file and parses as relay module
+    //  2. Converts the relay module to glenside
+    //  3. Generates random input (uniform [0, 255]) and parameters (uniform [-1, 1])
+    //  4. Runs relay module through TVM and benchmarks running glenside expr through interpreter
+    //  5. Compare output of using TVM vs interpreter
+    // $test_name: the name of the created test
+    // $relay_str_path: the path of the file containing the Relay code
+    macro_rules! test_model {
```

Comment:
Is this the same as `benchmark_model`, except that it doesn't benchmark? Do you think we need both?

Reply:
I think we need both if we want benchmark tests for the shallow models,
and still do full model testing (which currently takes a few hours)

Reply:
I'm not sure I understand why you can't use benchmark_model for the full models, though. Is it because of the `#[ignore]` that you're putting on the test in `test_model`? If that's the problem, take a look at how [egg uses macros to match on #[..] expressions](https://docs.rs/egg/0.3.0/egg/macro.define_language.html). In that case, you could match on #[..] expressions and attach them to the test. Then you could do 
```rust
#[ignore = "too slow"]
benchmark_model!(...full mobilenet...);
```

Reply:
hmm i tried this before and it didnt work

Reply:
ah thats what i was looking for thanks!